janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -151,6 +151,10 @@
             new ConfigOption<>(ES_CREATE_NS, "use-external-mappings",
             "Whether JanusGraph should make use of an external mapping when registering an index.", ConfigOption.Type.MASKABLE, false);
 
+    public static final ConfigOption<Boolean> USE_DYNAMIC_TEMPLATE =
+            new ConfigOption<>(ES_CREATE_NS, "use-dynamic-template",
+            "Whether JanusGraph should use a dynamic template when registering an index.", ConfigOption.Type.MASKABLE, false);
```

Comment:
"when registering a new **property*". Also add "Only applicable when 'use-external-mappings' is true"